packages/libs/react-ui/src/components/NavAccordion/NavAccordion.tsx

```diff
@@ -0,0 +1,36 @@
+'use client';
+
+import type { INavAccordionLinkProps, INavAccordionSectionProps } from '.';
+import type { NavAccordionState } from './NavAccordion.context';
+import {
+  initialOpenSections,
+  NavAccordionContext,
+} from './NavAccordion.context';
+import { navAccordionWrapperClass } from './NavAccordion.css';
+
+import type { FC, FunctionComponentElement } from 'react';
+import React, { useState } from 'react';
+
+type Child = FunctionComponentElement<
+  INavAccordionSectionProps | INavAccordionLinkProps
+>;
+export interface INavAccordionRootProps {
+  children?: Child[];
+  linked?: boolean;
+}
+
+export const NavAccordionRoot: FC<INavAccordionRootProps> = ({
+  children,
+  linked = false,
+}) => {
+  const [openSections, setOpenSections] =
+    useState<NavAccordionState>(initialOpenSections);
+
```

Comment:
I think you can put this in the context file so that this file is simpler and its collocated with the context

Reply:
I think this is the usual way of doing it unless you want to move to using actions and reducers which seems overly complex for what we're doing here.

Reply:
ahh ok, I think I'm thinking of that indeed. This looks good 👍